src/backend/controllers/emailControllers.ts

```diff
@@ -197,8 +200,57 @@ export async function sendNewBookingNotificationToStaffs(booking: ApiBooking) {
   });
 }
 
-export async function sendEmailWithDynamicTemplate(
-  messageData: SendDynamicEmailParams,
+/**
+ * Send a notification to the original sender of a letter.
+ *
+ * This method does NOT check if there is a published reply, it must only be called if there is.
+ *
+ * But it DOES check whether the customer has given their email address. If not, will return undefined.
+ *
+ * @param letterUuid
+ * @returns True if sent successfully, false if an error occured while sending, undefined if didn't even try to send.
+ */
+export async function sendReplyNotificationToCustomer(letterUuid: string): Promise<
+  | { sent: true }
+  | {
+      sent: false;
+      reason: 'no from email' | 'no notification template' | 'no customer email' | 'sendgrid error';
+    }
+> {
+  const { sendGridFromEmailAddress, sendGridCustomerReplyNotificationTemplate } = getConfig();
+  if (!sendGridFromEmailAddress) {
+    console.log('No From email adress was set');
+    return { sent: false, reason: 'no from email' };
+  }
+  if (!sendGridCustomerReplyNotificationTemplate) {
+    console.log('No notification template was set');
+    return { sent: false, reason: 'no notification template' };
+  }
+
+  const customerEmail = await getLetterCustomerEmailByUuid(letterUuid);
+  if (!customerEmail) {
+    return { sent: false, reason: 'no customer email' };
+  }
+
+  const success = await sendEmailWithDynamicTemplate<Record<string, never>>({
+    to: customerEmail,
+    from: {
+      name: 'Reply Notifcation',
```

Comment:
A typo here, also should we have email translation for the supported language?

Reply:
A challenge with translating this is that there is no easy way to know which language the user wants to get the email in. For staff/volunteer users we would need to store the language in their profile information, and for customers we would need to record which version of the letter writing form they used to send the letter.

Another options could be to make this configurable via an environment variable just like the address below it, I think I will do that so that Naisten Linja can modify this more easily in the future.

Reply:
These are fixed now, I also added the config variables to Heroku and tested locally that the emails still get sent as expected. This should be fine to merge now